protocols/indexer/contracts/Indexer.sol

```diff
@@ -193,28 +193,24 @@ contract Indexer is IIndexer, Ownable {
     *
     * @param _signerToken address
     * @param _senderToken address
-    * @param _count uint256
+    * @param _startAddress address The address to start from in the list of intents
+    * @param _count uint256 The total number of intents to return
     * @return locators address[]
     */
   function getIntents(
     address _signerToken,
     address _senderToken,
+    address _startAddress,
     uint256 _count
   ) external view returns (
     bytes32[] memory locators
   ) {
-
     // Ensure neither token is blacklisted.
-    if (!blacklist[_signerToken] && !blacklist[_senderToken]) {
-
-      // Ensure the index exists.
-      if (indexes[_signerToken][_senderToken] != Index(0)) {
-
-        // Return an array of locators for the index.
-        return indexes[_signerToken][_senderToken].fetchLocators(_count);
-
-      }
+    if (!blacklist[_signerToken] && !blacklist[_senderToken] && indexes[_signerToken][_senderToken] != Index(0)) {
+      // Return an array of locators for the index.
+      return indexes[_signerToken][_senderToken].fetchLocators(_startAddress, _count);
```

Comment:
Sorry, I know this is me being picky, but I was thinking more of a fail first style. 

i.e. The first condition should return `bytes32[](0)` and where the valid case is at the end. This way if everything is valid the code falls through to pass.

Reply:
Will change

Reply:
![image](https://user-images.githubusercontent.com/34962750/67403646-3d9f9b80-f580-11e9-915e-ab5d915b7ea7.png)